docs/developing_iree/ccache.md

```diff
@@ -0,0 +1,55 @@
+# Using `ccache` to build IREE
+
+[`ccache`](https://ccache.dev/) is a compilation cache. In principle, just prepending compiler invocations with `ccache` is all one needs to enable it, e.g.
+```shell
+ccache clang foo.c -c -o foo.o
+```
+takes care of executing `clang` with these arguments and caches the output file `foo.o`. The next invocation then skips executing `clang` altogether.
+
+## Installing and setting up `ccache`
+
+`ccache` should be easy to install about anywhere. On Debian-based Linux distributions, do:
+```shell
+sudo apt install ccache
+```
+
+The one `ccache` setting that you probably need to configure is the maximum cache size. The default `5G` is too small for our purposes. To set the cache max size, do this once:
+```shell
+ccache --max-size=20G
+```
+
+**Tip:** At the moment (late 2020), most of the code we're building is `third_party/llvm-project` so the fundamental limiting factor to how far we can cache away rebuilds is how often that dependency gets updated. Given how frequently it current is update, I'm finding that `20G` is enough to make the `ccache` size not be the limiting factor.
+
+## Telling CMake to use `ccache`
+
+This is actually a complex topic, there are two different approaches here as explained in this [article](https://crascit.com/2016/04/09/using-ccache-with-cmake/).
+
+### Suggested approach: pass CMake options
+
+In the initial CMake configuration step, set the `IREE_ENABLE_CCACHE` and `LLVM_CCACHE_BUILD` options, like this:
+```shell
+cmake -G Ninja \
+  -DIREE_ENABLE_CCACHE=ON \
+  -DLLVM_CCACHE_BUILD=ON \
+  ... other options as usual
+```
+
+Notes:
+* In the above command line, I explicitly put the `-G Ninja` flag even though that's not a new flag, we're always using `Ninja`. The reason is that the whole approach that these options are relying on only works on certain generators --- `Ninja` and `Makefile`. So if we ever used different generators, we'd have to switch to the other approach, as explained in the above linked [article](https://crascit.com/2016/04/09/using-ccache-with-cmake/).
```

Comment:
We recommend Ninja, but do not require it.

```suggestion
* If you use a generator other than `Ninja` see the
  [article](https://crascit.com/2016/04/09/using-ccache-with-cmake/)
  for other approaches.
```

Reply:
thanks!